services/sources/dspace_service.py

```diff
@@ -0,0 +1,143 @@
+from datetime import datetime, timedelta, timezone
+import os
+import requests
+from io import BytesIO
+from lxml import etree
+from constants.get_constants import get_constants
+from logger import create_log
+from mappings.base_mapping import MappingError
+from managers import S3Manager, RabbitMQManager
+from mappings.xml import XMLMapping
+from .source_service import SourceService
+
+logger = create_log(__name__)
+
+
+class DSpaceService(SourceService):
+    OAI_NAMESPACES = {
+        'oai_dc': 'http://www.openarchives.org/OAI/2.0/oai_dc/',
+        'dc': 'http://purl.org/dc/elements/1.1/',
+        'datacite': 'https://schema.datacite.org/meta/kernel-4.1/metadata.xsd',
+        'oapen': 'http://purl.org/dc/elements/1.1/',
+        'oaire': 'https://raw.githubusercontent.com/rcic/openaire4/master/schemas/4.0/oaire.xsd'
+    }
+
+    def __init__(self, base_url, SourceMapping: XMLMapping):
+        self.s3_manager = S3Manager()
+        self.s3_manager.createS3Client()
+        self.s3_bucket = os.environ['FILE_BUCKET']
+
+        self.file_queue = os.environ['FILE_QUEUE']
+        self.file_route = os.environ['FILE_ROUTING_KEY']
+
+        self.rabbitmq_manager = RabbitMQManager()
+        self.rabbitmq_manager.createRabbitConnection()
+        self.rabbitmq_manager.createOrConnectQueue(
+            self.file_queue, self.file_route)
+
+        self.constants = get_constants()
+
+        self.base_url = base_url
+        self.SourceMapping = SourceMapping
```

Comment:
I think we can do source_mapping for both the param and class variable here. 

It looks a bit funky when you instantiate an object but it's common practice to use snake case.